src/status_im/extensions/ethereum.cljs

```diff
@@ -98,8 +98,119 @@
 
 (handlers/register-handler-fx
  :extensions/ethereum-call
- (fn [{db :db} [_ _ {:keys [to] :as arguments}]]
-   (wrap-with-resolution db arguments :to execute-ethcall)))
+ (fn [{db :db} [_ _ {:keys [address] :as arguments}]]
+   (wrap-with-resolution db arguments :address execute-ethcall)))
+
+(handlers/register-handler-fx
+ :extensions/ethereum-erc20-total-supply
+ (fn [{db :db} [_ _ {:keys [address] :as arguments}]]
+   (let [arguments' (merge arguments
+                           {:address address
```

Comment:
you're right. but to avoid confusion and clearly decouple event arguments from internal web3 call arguments, if you agree I create a new map for the internal web3 explicitly. or maybe using `select-keys`.